source/codegen/templates/service_helpers.mako

```diff
@@ -69,20 +77,30 @@ ${set_response_values(output_parameters)}\
 %>\
 ${initialize_input_params(function_name, non_ivi_params)}\
 ${initialize_output_params(scalar_output_parameters)}\
-      auto status = library_->${function_name}(${service_helpers.create_args_for_ivi_dance_with_a_twist(parameters)});
-      if (status < 0) {
-        response->set_status(status);
-        return ::grpc::Status::OK;
-      }
+      while (true) {
+        auto status = library_->${function_name}(${service_helpers.create_args_for_ivi_dance_with_a_twist(parameters)});
+        if (status < 0) {
+          response->set_status(status);
+          return ::grpc::Status::OK;
+        }
+<%block filter="common_helpers.indent(1)">\
 ${initialize_output_params(array_output_parameters)}\
-      status = library_->${function_name}(${service_helpers.create_args(parameters)});
-      response->set_status(status);
+</%block>\
+        status = library_->${function_name}(${service_helpers.create_args(parameters)});
+        if (status == kErrorReadBufferTooSmall) {
```

Comment:
SO ivi-dance-with-a-twist is always a too-small error (for non zero-but-too-small inputs) but plain-ivi is either error or new size (depends on driver, I guess)?

Can you spell that out in the PR description (and commit description)?

And I think you pointed me to your source on ivi-dance-with-a-twist, although I don't know if that's one example or intrinsic to the twist.

Reply:
updated the PR description